core/src/style/style.cpp

```diff
@@ -95,13 +95,20 @@ void Style::setupShaderUniforms(int _textureUnit, Scene& _scene) {
 
 void Style::onBeginDrawFrame(const View& _view, Scene& _scene, int _textureUnit) {
 
+    // TODO cache which uniforms are used by the shader!
```

Comment:
In my measurements the uniform hash table lookup became the most expensive part cpu-wise in Tangram::render() + the string allocations to pass the uniform name to setUniform(const std::string)

I think of a UniformHandle template could help to keep all infos on the caller site. Something like
```
template<T>
struct UniformHandle {
    int id;
    int generation;
    T value;
    std::string name;
    set(shader&, T&)
}
 ```